code/controllers/subsystem/assets.dm

```diff
@@ -6,9 +6,10 @@ SUBSYSTEM_DEF(assets)
 	var/list/preload = list()
 
 /datum/controller/subsystem/assets/Initialize(timeofday)
-	for(var/type in typesof(/datum/asset) - list(/datum/asset, /datum/asset/simple))
-		var/datum/asset/A = new type()
-		A.register()
+	for(var/type in typesof(/datum/asset))
```

Comment:
Should this not be at least subtypesof?

Reply:
The `_abstract` check handles it here, skipping `/datum/asset` and other which aren't meant to be instantiated. Checking `/datum/language` is slightly redundant but it doesn't cause any harm because it doesn't pass the `icon != 'icons/misc/language.dmi'` check.